File: HslCommunication_Java/HslCommunication/src/HslCommunication/Profinet/Omron/OmronFinsDataType.java
package HslCommunication.Profinet.Omron;

/**
 * 欧姆龙的Fins协议的数据类型
 */
public class OmronFinsDataType
{
    /**
     * 实例化一个Fins的数据类型
     * @param bitCode 位操作的指令
     * @param wordCode 字操作的指令
     */
    public OmronFinsDataType( byte bitCode, byte wordCode )
    {
        BitCode = bitCode;
        WordCode = wordCode;

    }

    /**
     * 获取进行位操作的指令
     * @return
     */
    public byte getBitCode() {
        return BitCode;
    }

    /**
     * 进行字操作的指令
     * @return
     */
    public byte getWordCode() {
        return WordCode;
    }


    private byte BitCode = 0;
    private byte WordCode = 0;



    /**
     * DM Area
     */
    public static final OmronFinsDataType DM = new OmronFinsDataType( (byte) 0x02, (byte) 0x82 );

    /**
     * CIO Area
     */
    public static final OmronFinsDataType CIO = new OmronFinsDataType(  (byte)0x30,  (byte)0xB0 );

    /**
     * Work Area
     */
    public static final OmronFinsDataType WR = new OmronFinsDataType(  (byte)0x31,  (byte)0xB1 );

    /**
     * Holding Bit Area
     */
    public static final OmronFinsDataType HR = new OmronFinsDataType(  (byte)0x32,  (byte)0xB2 );

    /**
     * Auxiliary Bit Area
     */
    public static final OmronFinsDataType AR = new OmronFinsDataType(  (byte)0x33,  (byte)0xB3 );
}

File: HslCommunication_Net45/Profinet/Omron/OmronFinsDataType.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace HslCommunication.Profinet.Omron
{
    /// <summary>
    /// 欧姆龙的Fins协议的数据类型
    /// </summary>
    public class OmronFinsDataType
    {
        /// <summary>
        /// 实例化一个Fins的数据类型
        /// </summary>
        /// <param name="bitCode">进行位操作的指令</param>
        /// <param name="wordCode">进行字操作的指令</param>
        public OmronFinsDataType( byte bitCode, byte wordCode )
        {
            BitCode = bitCode;
            WordCode = wordCode;
        }



        /// <summary>
        /// 进行位操作的指令
        /// </summary>
        public byte BitCode { get; private set; }

        /// <summary>
        /// 进行字操作的指令
        /// </summary>
        public byte WordCode { get; private set; }



        /// <summary>
        /// DM Area
        /// </summary>
        public static readonly OmronFinsDataType DM = new OmronFinsDataType( 0x02, 0x82 );
        /// <summary>
        /// CIO Area
        /// </summary>
        public static readonly OmronFinsDataType CIO = new OmronFinsDataType( 0x30, 0xB0 );
        /// <summary>
        /// Work Area
        /// </summary>
        public static readonly OmronFinsDataType WR = new OmronFinsDataType( 0x31, 0xB1 );
        /// <summary>
        /// Holding Bit Area
        /// </summary>
        public static readonly OmronFinsDataType HR = new OmronFinsDataType( 0x32, 0xB2 );
        /// <summary>
        /// Auxiliary Bit Area
        /// </summary>
        public static readonly OmronFinsDataType AR = new OmronFinsDataType( 0x33, 0xB3 );
    }
}
